documentation/add/09_Serializable_Interfaces.md

```diff
@@ -0,0 +1,132 @@
+
+Two remaining methods presented on the **Fig 1** namely static `loadobj` and `from_old_struct` methods need overloading if/when you want to support loading the versions of a class, different from the the current class version. By default, a standard `loadobj` method exists and this method expects to receive the structure, produced by `to_struct` method. `to_struct` method adds information about the class name, so generic `loadobj` extracts this information and recovers the class. If this information is missing, which may happen when you loading a class structure, generated before the object become `serializable`, you need to overload `loadobj` to provide the empty instance of the class, as described by the code snippet below:
+```Matlab
+        methods(Static)
+            function obj = loadobj(S)
+                % boilerplate loadobj method, calling generic method of
+                % saveable class. Put it as it is replacing the
+                "ChildClass" name by the name of the class the loadobj is
+                the method of.
+                obj = ChildClass();
+                obj = loadobj@serializable(S,obj);
+            end
+        end
+```
+By default, the method `from_old_struct` calls `from_bare_struct` method, so it does not need overloading if the structure of your class public interface have not changed.
+If it have changed, `from_old_struct` needs overloading with the code, converting the old information, stored in the previous class structure into the information, necessary to define new class.
+
+## Interdependent properties problem.
+
+All serializable interface described above and used inside the code is build under assumption that you define class value by assigning the values of the properties one by one. 
+All public setters run checks of the validity of the values one wants to assign to the properties and throw exceptions if the input values are invalid. This assumption causes one problem with interdependent properties. For example, arrays `s`,`e`, and `npix` of a `dnd` or `sqw` object need to be the same shape and size arrays. If you are modifying, say, empty `dnd` object assigning `s`, `e` and `npix` arrays one after another, the first assignment (for `s`) would be always invalid, as the sizes of two other arrays are different from the first one as the initial object was containing empty data in arrays `e` and `npix`.
+
+There is number of ways, one can deal with this issue. After discussion, we have decided to select the following approach, summarized on the **Fig 2** describing the interface  used to validate interdependent properties. 
+
+<center>![serializable Validation Interface](../diagrams/Serializable_validation_interface.png)</center>
+
+**Fig 2:** Validation interface for **serializable** class.
```

Comment:
Our normal naming convention is that A is a public property and A_ is its private implementation. Here check_combo_arg_ is a logical encapsulating the state that check_combo_arg looks at. I think the names should change to reflect the different function

Reply:
the idea was not to have public property `check_combo_arg` for protected property `check_combo_arg_`. But may be it makes sense to have one (hidden). Then the renaming is justified